src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -66,8 +68,8 @@ internal partial class DiagnosticIncrementalAnalyzer : IIncrementalAnalyzer2
 
         internal DiagnosticAnalyzerInfoCache DiagnosticAnalyzerInfoCache => _diagnosticAnalyzerRunner.AnalyzerInfoCache;
 
-        public bool IsCompilationEndAnalyzer(DiagnosticAnalyzer diagnosticAnalyzer, Project project, Compilation compilation)
-            => DiagnosticAnalyzerInfoCache.IsCompilationEndAnalyzer(diagnosticAnalyzer, project, compilation) == true;
+        public async Task<bool> IsCompilationEndAnalyzerAsync(DiagnosticAnalyzer diagnosticAnalyzer, Project project, CancellationToken cancellationToken)
+            => await DiagnosticAnalyzerInfoCache.IsCompilationEndAnalyzerAsync(diagnosticAnalyzer, project, cancellationToken).ConfigureAwait(false) == true;
```

Comment:
my point is more that you can just elide the `==true` as it seems like `IsCompilationEndAnalyzerAsync` returns a bool, right?

Reply:
No, it returns a `bool?`. The unclear path is when analyzer throws an exception during the callback. I have filed https://github.com/dotnet/roslyn/issues/47754 to get rid of this error prone logic completely.